### PR TITLE
RAIL-14 GET /journey/:id operable

### PR DESCRIPTION
**Pull request by orzeuek**

### CODE REVIEW

File: src/service/PersistanceService.ts [hunk 1]
```diff
@@ -56,39 +59,52 @@ export class FareOptionCache {
...
         }
+    }
 
+    /**
+     * Persist fare option in this.cache.
+     * key = fareOption.id
+     * value = strigified fareOption object (complex objects are stored as a links!)
+     */
+    public async set(fareOption:FareOption): Promise<boolean> {
+        const strigifiedFareOption = JSON.stringify(FareOptionViewFactory(fareOption));
```
Comment (Dyktus) [inline src/service/PersistanceService.ts]: just name shouldn't it be `stringified`?

File: src/service/PersistanceService.ts [hunk 2]
```diff
@@ -56,39 +59,52 @@ export class FareOptionCache {
...
+        return this.cache.set(fareOption.id, strigifiedFareOption);
```
Comment (Dyktus) [inline src/service/PersistanceService.ts]: shouldn't it be `return await this.cache.set` ? I see that Cache.set returns Promise<boolean>

**Comment by orzeuek** [comment]
Linus, your review is still welcome :)

File: src/service/PersistanceService.ts [hunk 3]
```diff
@@ -56,39 +59,52 @@ export class FareOptionCache {
...
-                    fares: fareOption.fares.map(f => {
-                        return {
-                            adults: f.adults,
-                            children: f.children,
-                            fare: f.fare.id
-                        };
-                    })
-                });
-                this.cache.set(fareOption.id, strigifiedFareOption).catch((err) => {
+                this.set(fareOptionOption.get).catch((err) => {
```
Comment (linusnorton) [inline src/service/PersistanceService.ts]: maybe an await, would be better. That throw inside the promise callback won't do anything apart from generate an unhandled error exception

Comment (orzeuek) [reply src/service/PersistanceService.ts]: done in: https://github.com/assertis/odyssey-service/commit/e2f28349a95fdb80ff8e645f3e8bf583e0139f72 - thanks

**Pull request merged**